### ISSUE TITLE
Toggle write access for connected relays

### ISSUE THREAD

**Issue by badonyx** [damus-io/damus#1605]
I want to connect to some relays but not write to them.

**Comment by alltheseas** [comment]
Is this a duplicate of https://github.com/damus-io/damus/issues/453

**Comment by badonyx** [comment]
No but it overlaps with #525

**Comment by alltheseas** [comment]
> I want to connect to some relays but not write to them.

Can you explain to me what you see yourself doing immediately before, and during this job-to-be-done, and why?

**Comment by badonyx** [comment]
I'm not entirely sure what you mean. This is a standard feature in many clients. An example might be if I want to see content from the activitypub bridge relay `relay.mostr.pub` but I don't care to write my comments there to interact with activitypub users.

**Comment by alltheseas** [comment]
> Example use cases:

These are various use cases. It helps the team a lot for the proposed scope to be very well defined, with clear context & benefits.

Can you rank from the most important to the least important and add any further context on why these are valuable for you?

**Comment by badonyx** [comment]
I don't know what language you are speaking. This is one clear and concise feature. I listed some examples of how it would be useful, or how it is already useful in other clients.

If you read [NIP-65](https://github.com/nostr-protocol/nips/blob/master/65.md) or look at any kind-3 event it might help to understand what is meant by `read/write`.

```
{
  "id": ...,
  "pubkey": ...,
  "content": "{\"wss://relay.damus.io/\":{\"write\":true,\"read\":true},...}",
  "sig": ...,
  "kind": 3,
  "created_at": ...,
  "tags": [
...
```

**Comment by jb55** [comment]
On Wed, Oct 18, 2023 at 12:21:11PM -0700, badonyx wrote:
>I don't know what language you are speaking. This is one clear and concise feature. I listed some examples of how it would be useful, or how it is already useful in other clients.
>
>If you read [NIP-65](https://github.com/nostr-protocol/nips/blob/master/65.md) or look at any kind-3 event it might help to understand what is meant by `read/write`.
>
>```
>{
>  "id": ...,
>  "pubkey": ...,
>  "content": "{\"wss://relay.damus.io/\":{\"write\":true,\"read\":true},...}",
>  "sig": ...,
>  "kind": 3,
>  "created_at": ...,
>  "tags": [
>...
>```

Thanks for the condescending remark. I am quite aware of these fields,
as Damus was the first mobile client and I needed to parse these. Damus
has never used them because they never seemed that useful, or they were
too complex for the regular user to manage.

Most users don't want to think about relays let alone the read/write
state on relays.

I think it makes more sense to make writing contextual. For example,
clicking on a relay opens up a discord-like view of the relay, and
reading and writing gets scoped to only that relay when you have this
view open. I am more interested in these kinds of user experiences
rather than flipping obscure relay settings.

**Comment by badonyx** [comment]
I agree that those read/write fields in kind 3 events are not a good implementation. It would be better to discontinue support for relay lists in kind 3 events and instead use NIP-65 / kind 10002, which better implements the read/write fields.

> Thanks for the condescending remark. I am quite aware of these fields, as Damus was the first mobile client and I needed to parse these.

I don't know what this means. I was talking to @alltheseas who did not seem to understand the context.

**Comment by alltheseas** [comment]
> I was talking to @alltheseas who did not seem to understand the context, and their questions did not make sense to me.

@badonyx here's what I'm getting at. You write a feature request. 

I am trying to figure why you need this request through a product lens. 

If other clients are doing this or if there is a NIP, great.

The team and I take a view of the thousands of the requests we get, and filter for how useful and what are the benefits of this feature. 

If a request is evaluated as improving the user experience, improving retention, adding a feature that many if not all will benefit from etc.. we will strongly consider.

If there is a NIP or client X is doing something it does *not* mean we will implement.

I was and am not convinced (and sounds like neither is Will) that this feature is for Damus users. 

My questions to you were an attempt at asking you to educate and convince me why Damus should add this.

**Comment by alltheseas** [comment]
> I think it makes more sense to make writing contextual. For example,
> clicking on a relay opens up a discord-like view of the relay, and
> reading and writing gets scoped to only that relay when you have this
> view open. I am more interested in these kinds of user experiences
> rather than flipping obscure relay settings.

https://github.com/damus-io/damus/issues/1624

**Comment by alltheseas** [comment]
Closing as part of a backlog triage — we're narrowing focus to the most actionable issues to reduce noise for contributors.

If this is still relevant, please reopen with updated context or reproduction steps. Thanks for reporting!
